x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchTests.java

```diff
@@ -296,6 +298,38 @@ public void testParserBadActions() throws Exception {
         }
     }
 
+    public void testParserConsumesEntireDefinition() throws Exception {
+        WatchParser wp = createWatchparser();
+        String watchDef="{\n" +
```

Comment:
Typically we use XContentBuilder to build the json (see the test below for an example).

Reply:
You can use XContentBuilder to build valid json, and then append an extra token (eg comma as you have). This will ensure there isn't malformed json in the middle which tricks the test.